TypeScript compile error `TS1479` even if you import `option-t/lib/*` or `option-t/cjs/*` in a project set `moduleResolution=node16` & `type=commonjs`

## Steps to Reproduce

### Step 0

Set up a project containing these files.

#### /package.json

```json
{
    "type": "commonjs",
    "scripts": {
        "build": "tsc"
    },
  "devDependencies": {
    "typescript": "^4.9.4"
  }
}
```

#### /src/bar.ts

```ts
import {} from `option-t/lib/Nullable/Nullable`;
```

#### tsconfig.json

```json5
{
    "moduleResolution": "Node16",
}
```

### Step 1

Run `npm run build`



## Actual Result

tsc show this error

> error TS1479: The current file is a CommonJS module whose imports will produce 'require' calls; however, the referenced file is an ECMAScript module and cannot be imported with 'require'. Consider writing a dynamic 'import("option-t/lib/Nullable/Nullable")' call instead.
>  To convert this file to an ECMAScript module, change its file extension to '.mts' or create a local package.json file with `{ "type": "module" }`.

--- a/tools/package_json_rewriter/transformer/add_exports_field/ExportEntry.mjs
+++ b/tools/package_json_rewriter/transformer/add_exports_field/ExportEntry.mjs
@@ -284,16 +284,15 @@ function constructDualPackagePathValue({ cjs, esm, dts }) {
     //  * `["node", "import"]` is used as _defaultEnv_ for its ES Module resolver.
     //
     // see also https://nodejs.org/api/esm.html#esm_conditional_exports
-    return Object.freeze({
-        // > Note that the "types" condition should always come first in "exports".
-        // https://devblogs.microsoft.com/typescript/announcing-typescript-4-7/#package-json-exports-imports-and-self-referencing
-        'types': dts,
+    return {
         'import': esm,
         'require': cjs,
+        // https://devblogs.microsoft.com/typescript/announcing-typescript-4-5-beta/#packagejson-exports-imports-and-self-referencing
+        'types': dts,
         // _default_ should be placed to the last.
         // https://nodejs.org/api/esm.html#esm_conditional_exports
         'default': esm,
-    });
+    };
 }
 
 function constructPathValue({ filepath, dts }) {
@@ -305,11 +304,11 @@ function constructPathValue({ filepath, dts }) {
     //  * `["node", "import"]` is used as _defaultEnv_ for its ES Module resolver.
     //
     // see also https://nodejs.org/api/esm.html#esm_conditional_exports
-    return Object.freeze({
+    return {
         // https://devblogs.microsoft.com/typescript/announcing-typescript-4-5-beta/#packagejson-exports-imports-and-self-referencing
         'types': dts,
         // _default_ should be placed to the last.
         // https://nodejs.org/api/esm.html#esm_conditional_exports
         'default': filepath,
-    });
+    };
 }

